cpp/src/shim/cufile.cpp

```diff
@@ -79,21 +69,6 @@ cuFileAPI::cuFileAPI()
     get_symbol(StreamRegister, lib, KVIKIO_STRINGIFY(cuFileStreamRegister));
     get_symbol(StreamDeregister, lib, KVIKIO_STRINGIFY(cuFileStreamDeregister));
   }
-
-  // cuFile is supposed to open and close the driver automatically but
-  // because of a bug in cuFile v1.4 (CUDA v11.8) it sometimes segfaults:
-  // <https://github.com/rapidsai/kvikio/issues/159>.
-  if (version < 1050) { driver_open(); }
```

Comment:
@vuule could you please double check whether this is ok for you?

Asking as you had originally opened issue ( https://github.com/rapidsai/kvikio/issues/159 ). However based on the comment above, this should not be an issue in CUDA 12.0+

Reply:
Note: That KvikIO with newer cuFile versions handles driver opening/closing in this code

https://github.com/rapidsai/kvikio/blob/33a734cde6f4c5af5960fed826131e8797b5b46a/cpp/src/cufile/driver.cpp#L45-L56

https://github.com/rapidsai/kvikio/blob/33a734cde6f4c5af5960fed826131e8797b5b46a/cpp/src/shim/cufile.cpp#L108-L122